libavfilter/vf_livepeer_dnn.c

```diff
@@ -119,41 +84,45 @@ static av_cold int init(AVFilterContext *context)
 
     ret = ff_dnn_init(&ctx->dnnctx, DFT_PROCESS_FRAME, context);
 
-    // Pre-execute the model after loading the input/output dimensions to
-    // allocate the tensors before the first frame comes in
-    int ret2 = pre_exec(context);
-    if (ret2 < 0) {
-        av_log(ctx, AV_LOG_ERROR, "pre-exec for the model failed\n");
-        return ret2;
+    ctx->dnnctx.model->get_input(ctx->dnnctx.model->model, &input, ctx->dnnctx.model_inputname);
+    ctx->input_width = input.width;
+    ctx->input_height = input.height;
+    // pre-executes the model and gets output information
+    if (DNN_SUCCESS != ctx->dnnctx.model->get_output(ctx->dnnctx.model->model, ctx->dnnctx.model_inputname,
```

Comment:
Neat - I had no idea that `get_output` API did the execution we wanted under the hood!